pkg/web_app/lib/src/foldable.dart

```diff
@@ -22,41 +23,26 @@ void _setEventForFoldable() {
     final scrollContainer = _parentWithClass(h, 'scroll-container');
     if (content == null) continue;
 
-    Future<void> update(bool isActive) async {
-      // Closing is simple: no measurements, no scrolling.
+    Future<void> toggle() async {
+      final isActive = foldable.classes.toggle('-active');
       if (!isActive) {
-        content.style.maxHeight = '0px';
         return;
       }
 
-      /// The following coordinate and dimension measurements trigger a reflow,
-      /// but it is acceptable, as it is local to this event processing and the
-      /// impact is low.
-      if (content.style.display != 'block') {
-        content.style.display = 'block';
-      }
-      // Needs to be empty to measure real dimension.
-      content.style.maxHeight = '';
-
-      final contentHeight = content.offsetHeight;
-      final boundingRect = content.getBoundingClientRect();
-      final scrollContainerHeight = scrollContainer?.clientHeight;
-      final buttonHeight = h.offsetHeight;
-
-      // Reset content state as hidden.
-      content.style.maxHeight = '0px';
-
-      // Wait one animation frame before trigger the full height content.
-      await window.animationFrame;
-      content.style.maxHeight = '${contentHeight}px';
-
       num scrollDiff = 0;
       if (scrollContainer != null) {
+        // Wait one animation frame before measurements.
```

Comment:
Not sure I understand all this scrolling logic...

Do we really need it?

Reply:
This is for the mobile view: when the navigation menu is at the bottom, and it is clicked, there is no indication that something has been displayed under the fold. Scrolling to it makes the new content visible.

Reply:
Can we just use [scrollIntoView](https://caniuse.com/?search=scrollIntoView)?

Reply:
I think we have tried it and it worked with full-page elements, but had issues with embedded-scroll of the navigation. It may be worth to re-evaluate it again, a lot of time has passed since.

Reply:
https://github.com/dart-lang/pub-dev/issues/7342 to track it separately